janusgraph-core/src/main/java/org/janusgraph/graphdb/management/ConfigurationManagementGraph.java

```diff
@@ -0,0 +1,309 @@
+    public static final String PROPERTY_CREATED_USING_TEMPLATE = "Created_Using_Template";
+    private static final String VERTEX_LABEL = "Configuration";
+    private static final String GRAPH_NAME_INDEX = "Graph_Name_Index";
+    private static final String PROPERTY_TEMPLATE = "Template_Configuration";
+    private static final String TEMPLATE_INDEX = "Template_Index";
+
+    private final StandardJanusGraph graph;
+
+    /**
+     * This class allows you to create/update/remove configuration objects used to open a graph.
+     * To use these APIs, you must define one of your graph's key as "JanusGraphConfigurationGraph"
+     * in your server's YAML; the configuration objects you define using these APIs will be stored
+     * in a graph representation (on a vertex, more precisely), and this graph will be opened
+     * according to the file supplied along with the "JanusGraphConfigurationGraph" key.
+     */
+    // Would  prefer this constructor to have no modifier (for instantiation by JanusGraphManager,
+    // but for now, leaving open for testing purposes
+    public ConfigurationManagementGraph(StandardJanusGraph graph) {
+        initialize();
+        this.graph = graph;
+        createIndexIfDoesNotExist(GRAPH_NAME_INDEX, PROPERTY_GRAPH_NAME, String.class, true);
+        createIndexIfDoesNotExist(TEMPLATE_INDEX, PROPERTY_TEMPLATE, Boolean.class, false);
+        createIndexIfDoesNotExist(PROPERTY_CREATED_USING_TEMPLATE, PROPERTY_CREATED_USING_TEMPLATE, Boolean.class, false);
+    }
+
+    private synchronized void initialize() {
+        if (null != instance) {
+            final String errMsg = "ConfigurationManagementGraph should be instantiated just once, by the JanusGraphManager.";
+            throw new ConfigurationManagementGraphAlreadyInstantiatedException(errMsg);
+        }
+        this.instance = this;
+    }
+
+    /**
+     * If one of your "graphs" key was equivalent to \"JanusConfigurationGraph\" in your
+     * YAML file supplied at server start, then we return the ConfigurationManagementGraph
+     * Singleton-- otherwise we throw a {@link ConfigurationManagementGraphNotEnabledException} exception.
+     */
+    public static ConfigurationManagementGraph getInstance() throws ConfigurationManagementGraphNotEnabledException {
+        if (null == instance) {
+            throw new ConfigurationManagementGraphNotEnabledException(
+                "Please add a key named \"JanusConfigurationGraph\" to the \"graphs\" property " +
+                "in your YAML file and restart the server to be able to use the functionality " +
+                "of the ConfigurationManagementGraph class."
+            );
+        }
+
+        return instance;
+    }
+
+    /**
+     * Create a configuration according to the supplied {@link MapConfiguration}; you must include
+     * the property "graph.graphname" with a {@link String} value in the configuration; you can then
```

Comment:
Can we remove `{@link String}` here and throughout docs in the PR? I don't think this is done in javadoc elsewhere in the project and in reviewing the generated API docs here I think the String links distract from the rest of the documentation content (which is really great, by the way!).